Form/FormMapper.php

```diff
@@ -275,6 +280,21 @@ public function addHelp($name, $help)
     }
 
     /**
+     * Symfony default form class sadly can't handle
+     * form element with dots in its name (when data
+     * get bound, the default dataMapper is a PropertyPathMapper).
+     * So use this trick to avoid any issue.
+     *
+     * @param string $fieldName
+     *
+     * @return string
+     */
+    public function sanitizeFieldName($fieldName)
+    {
+        return str_replace(array('__', '.'), array('____', '__'), $fieldName);
```

Comment:
Yes, that was my point. With your actual changes, you are still getting this behaviour, but you have to do 2 replaces.

Anyway, in some place of the code, this is getting a replace to get the correct path right (move from __ to .)? It is only on `ChoiceFieldMaskType`?